main.cpp

```diff
@@ -102,19 +117,45 @@ bool is_created(const string &filename){
     return false;
 }
 
-int main() {
+void parse_commandline_args(int argc, char** &argv)
+{
+    struct option longOpts[] = {
+            { "help", no_argument, nullptr, 'h' },
+            { nullptr, 0, nullptr, '\0' }
+        };
+
+    int option, optionIndex;
+
+    while((option = getopt_long(argc, argv, "h", longOpts, &optionIndex))!= -1)
+    {
+        switch(option)
+        {
+            case 'h':{
+            
+                cout<<help;
+            }
+                break;
+            default:
+                break;
+        }
+    }
+}
+
+int main(int argc, char ** argv) {
     /*
-     * 1) get the cpp files
-     * 2) determine what cpp files to recompile
+     * 1) parse for commandline arguments
+     * 2) get the cpp files
+     * 3) determine what cpp files to recompile
      *      - need to recompile if no obj file we have to create it, or if there is an obj file,
      *      but the cpp file has been modified more recently than the obj file
      *      - need to check to see if there is an executable created
-     * 3) if none, we are done
-     * 4) else, recompile them (done!)
-     * 5) relink everything
-     * 6) done!
+     * 4) if none, we are done
+     * 5) else, recompile them (done!)
+     * 6) relink everything
+     * 7) done!
      */
     try {
+        parse_commandline_args(argc, argv); 
```

Comment:
Probably should add a way for this to return a boolean or something. If we get a help message or if someone requests some option that doesn't exist we would want to exit without error instead of going through work